aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -556,8 +563,13 @@ void bgemm_internal_cublas<c10::complex<float>>(CUDABLAS_BGEMM_ARGTYPES(c10::com
       reinterpret_cast<cuComplex*>(c), ldc, stridec, num_batches));
 }
 
-template <>
-void bgemm_internal_cublas<at::Half>(CUDABLAS_BGEMM_ARGTYPES(at::Half)) {
+template <typename C_Dtype>
+void bgemm_internal_cublas_half_helper(CUDABLAS_BGEMM_ARGTYPES_AND_C_DTYPE(at::Half, C_Dtype)) {
+  #ifdef USE_ROCM
+  if (std::is_same_v<C_Dtype, float>)
+    TORCH_CHECK(false, "bgemm input type at::Half and output type float is not supported for ROCm");
```

Comment:
cc @jeffdaily can someone double check if these combinations are supported in RocBLAS(Lt)?

Reply:
Looks like no.  https://rocm.docs.amd.com/projects/hipBLASLt/en/develop/reference/data-type-support.html#supported-data-types